include/ensmallen_bits/spsa/spsa.hpp

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file spsa.hpp
+ * @author N Rajiv Vaidyanathan
+ * @author Marcus Edel
+ *
+ * SPSA (Simultaneous perturbation stochastic approximation) method for
+ * faster convergence.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_SPSA_SPSA_HPP
+#define ENSMALLEN_SPSA_SPSA_HPP
+
+namespace ens {
+
+/**
+ * Implementation of the SPSA methodT. The SPSA algorithm approximates the
+ * gradient of the function by finite differences along stochastic directions.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @article{Spall1998,
+ *   author  = {Spall, J. C.},
+ *   title   = {An Overview of the Simultaneous Perturbation Method for
+ *              Efficient Optimization},
+ *   journal = {Johns Hopkins APL Technical Digest},
+ *   volume  = {19},
+ *   number  = {4},
+ *   pages   = {482--492},
+ *   year    = {1998}
+ * }
+ * @endcode
+ *
+ */
+class SPSA
+{
+ public:
+  /**
+   * Construct the SPSA optimizer with the given function and parameters.  The
+   * defaults here are not necessarily good for the given problem, so it is
+   * suggested that the values used be tailored to the task at hand.  The
+   * maximum number of iterations refers to the maximum number of points that
+   * are processed (i.e., one iteration equals one point; one iteration does not
+   * equal one pass over the dataset).
+   *
+   * @param alpha Scaling exponent for the step size.
+   * @param batchSize Batch size to use for each step.
+   * @param gamma Scaling exponent for evaluation step size.
+   * @param a Scaling parameter for step size.
```

Comment:
It might be useful to go with a more descriptive parameter name (``a/c``) something like ``EvaluationStepSize`` and ``StepSize``.

Reply:
In the actual paper, the variables `a` and `c` are used. So, I have renamed them in the code as you have suggested and also mentioned about `a` and `c` in brackets for correlating with the paper. Is that fine?

Reply:
Sounds good for me.